HW11/Selenium.cs

```diff
@@ -0,0 +1,463 @@
+            IWebElement notFoundLink = _driver.FindElement(By.Id("invalid-url"));
+
+            Assert.IsTrue(homeLink.Displayed);
+            Assert.IsTrue(homehZvluLink.Displayed);
+            Assert.IsTrue(createdLink.Displayed);
+            Assert.IsTrue(noContentLink.Displayed);
+            Assert.IsTrue(movedLink.Displayed);
+            Assert.IsTrue(badRequestLink.Displayed);
+            Assert.IsTrue(unauthorizedLink.Displayed);
+            Assert.IsTrue(forbiddenLink.Displayed);
+            Assert.IsTrue(notFoundLink.Displayed);
+
+            homeLink.Click();
+            IReadOnlyCollection<string> windowHandles = _driver.WindowHandles;
+            _driver.SwitchTo().Window(windowHandles.Last());
+            Assert.AreEqual("https://demoqa.com/", _driver.Url);
+
+            _driver.SwitchTo().Window(windowHandles.First());
+
+            homehZvluLink.Click();
+            windowHandles = _driver.WindowHandles;
+            _driver.SwitchTo().Window(windowHandles.Last());
+            Assert.AreEqual("https://demoqa.com/", _driver.Url);
+
+            _driver.SwitchTo().Window(windowHandles.First());
+
+            createdLink.Click();
+
+            _driverWait.Until(drv =>
+                   {
+                       if (drv.FindElements(By.XPath("//*[@id=\'linkResponse\']")).Count > 0)
+                       {
+                           return true;
+                       }
+                       return false;
+                   });
+
+            var createdLinkStatusCode = _driver.FindElement(By.XPath("//p[@id='linkResponse']/b[.='201']"));
+            var createdLinkStatusText = _driver.FindElement(By.XPath("//p[@id='linkResponse']/b[.='Created']"));
+            
+            Assert.AreEqual("201", createdLinkStatusCode.Text);
+            Assert.AreEqual("Created", createdLinkStatusText.Text);
+
+            noContentLink.Click();
+
+            _driverWait.Until(drv =>
+            {
+                if (drv.FindElements(By.XPath("//p[@id='linkResponse']/b[.='204']")).Count > 0)
+                {
+                    return true;
+                }
+                return false;
+            });
+
+            var noContentLinkStatusCode = _driver.FindElement(By.XPath("//p[@id='linkResponse']/b[.='204']"));
+            var noContentLinkStatusText = _driver.FindElement(By.XPath("//p[@id='linkResponse']/b[.='No Content']"));
+            
+            Assert.AreEqual("204", noContentLinkStatusCode.Text);
+            Assert.AreEqual("No Content", noContentLinkStatusText.Text);
+
+            movedLink.Click();
+
+            _driverWait.Until(drv =>
+            {
+                if (drv.FindElements(By.XPath("//p[@id='linkResponse']/b[.='301']")).Count > 0)
+                {
+                    return true;
+                }
+                return false;
+            });
+
+            var movedLinkStatusCode = _driver.FindElement(By.XPath("//p[@id='linkResponse']/b[.='301']"));
```

Comment:
you have all the same logic for all contents you're checking, so you can create the only method and reuse it for all cases